sdk/cosmosdb/cosmos/samples-dev/ChangeFeedIterator/ChangeFeedIteratorAllVersionsAndDeletes.ts

```diff
@@ -0,0 +1,144 @@
+      paths: ["/name"],
+      version: PartitionKeyDefinitionVersion.V1,
+    },
+    throughput: 11000,
+  };
+  try {
+    const { container } = await database.containers.createIfNotExists(containerDef);
+    console.log(`Created container with id : ${containerId}`);
+
+    logStep("Change Feed for Entire Container");
+    let options: ChangeFeedIteratorOptions = {
+      maxItemCount: 1,
+      changeFeedStartFrom: ChangeFeedStartFrom.Now(),
+      changeFeedMode: ChangeFeedMode.AllVersionsAndDeletes,
+    };
+
+    await iterateChangeFeedFromNowAndContinuation(container, options, 1, 4);
+
+    logStep("Change Feed for a Partition Key");
+    options = {
+      maxItemCount: 1,
+      changeFeedStartFrom: ChangeFeedStartFrom.Now("sample1"),
+      changeFeedMode: ChangeFeedMode.AllVersionsAndDeletes,
+    };
+    await iterateChangeFeedFromNowAndContinuation(container, options, 5, 8);
+
+    logStep("Change Feed for an epk range");
+    const feedRanges = await container.getFeedRanges();
+    options = {
+      maxItemCount: 1,
+      changeFeedStartFrom: ChangeFeedStartFrom.Now(feedRanges[0]),
+      ChangeFeedMode: ChangeFeedMode.AllVersionsAndDeletes,
+    };
+    await iterateChangeFeedFromNowAndContinuation(container, options, 9, 12);
+  } catch (err) {
+    console.error(err);
+  } finally {
+    await finish();
+  }
+}
+
+async function ingestData(container: Container, initialize: number, end: number) {
+  for (let i = initialize; i <= end; i++) {
+    await container.items.create({ id: `item${i}`, name: `sample1`, key: i });
+    await container.items.create({ id: `item${i}`, name: `sample2`, key: i });
+    await container.items.create({ id: `item${i}`, name: `sample3`, key: i });
+    await container.items.create({ id: `item${i}`, name: `sample4`, key: i });
+  }
+  console.log(`ingested items with id - item${initialize} and id - item${end}`);
+}
+
+async function insertAndModifyData(container: Container, initialize: number, end: number) {
+  await ingestData(container, initialize, end);
+  await container.items.upsert({ id: `item${initialize}`, name: `sample1`, key: initialize + 1 });
+  console.log(`upserted item with id - item${initialize} and partition key - sample1`);
+  await container.item(`item${initialize}`, `sample1`).delete();
+  console.log(`deleted item with id - item${initialize} and partition key - sample1`);
+}
+
+async function iterateChangeFeedFromNowAndContinuation(
+  container: Container,
+  options: ChangeFeedIteratorOptions,
+  initialize: number,
+  end: number,
+): Promise<void> {
+  let iterator = container.items.getChangeFeedIterator(options);
+  console.log("running the iterator to start fetching changes from now.");
+  await iterator.readNext();
+  // ingest, upsert, and delete some data to introduce changes to container
+  await insertAndModifyData(container, initialize, initialize + 1);
+  // fetch the changes, and continuation token so that we can start from the same point in time
```

Comment:
Are we fetching the changes here?

Reply:
Maybe we can add a detailed log to explain why continuation token is required here.

Reply:
yes, we are fetching the changes and saving the continuation token to fetch changes from continuation token in next iteration.